sbi/simulators/simutils.py

```diff
@@ -266,6 +275,20 @@ def check_prior_methods(prior):
             with prior.log_prob(theta). Consider using a PyTorch distribution."""
         )
 
+    assert hasattr(prior, "mean"), "the prior must have a mean attribute or property."
```

Comment:
Or these could be in a different function. But this could also be the place where instead of complaining, one adds the attribute or method based on the fact that we know that `.sample` exists?

Reply:
to be discussed.